cli/flags.go

```diff
@@ -158,9 +160,11 @@ func initFlags(ctx *server.Context) {
 		}
 	}
 
+	isEnvRoachDev := os.Getenv("COCKROACH_DEV") != ""
```

Comment:
This will enable dev mode if `COCKROACH_DEV=0` or `COCKROACH_DEV=false` which is confusing. I think it would be better to explicitly state the value that enables dev-mode.